Apps/W1/HybridGP/app/src/Migration/Customers/GPCustomerMigrator.codeunit.al

```diff
@@ -375,4 +375,86 @@ codeunit 4018 "GP Customer Migrator"
+        if not MigrateCustomerClasses then
+            exit;
+
+        // Create the Customer Posting Groups
+        repeat
+            Clear(CustomerPostingGroup);
+            ClassId := GPRM00201.CLASSID.Trim();
+            if ClassId <> '' then
+                if not CustomerPostingGroup.Get(ClassId) then begin
+                    CustomerPostingGroup.Validate("Code", ClassId);
+                    CustomerPostingGroup.Validate("Description", GPRM00201.CLASDSCR);
+
+                    // Receivables Account
+                    AccountNumber := HelperFunctions.GetGPAccountNumberByIndex(GPRM00201.RMARACC);
+                    if AccountNumber <> '' then begin
+                        HelperFunctions.EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                        CustomerPostingGroup.Validate("Receivables Account", AccountNumber);
+                    end;
+
+                    // Payment Disc. Debit Acc.
+                    AccountNumber := HelperFunctions.GetGPAccountNumberByIndex(GPRM00201.RMTAKACC);
+                    if AccountNumber <> '' then begin
+                        HelperFunctions.EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                        CustomerPostingGroup.Validate("Payment Disc. Debit Acc.", AccountNumber);
+                    end;
+
+                    // Additional Fee Account
+                    AccountNumber := HelperFunctions.GetGPAccountNumberByIndex(GPRM00201.RMFCGACC);
+                    if AccountNumber <> '' then begin
+                        HelperFunctions.EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                        CustomerPostingGroup.Validate("Additional Fee Account", AccountNumber);
+                    end;
+
+                    // Payment Disc. Credit Acc.
+                    AccountNumber := HelperFunctions.GetGPAccountNumberByIndex(GPRM00201.RMAVACC);
+                    if AccountNumber <> '' then begin
+                        HelperFunctions.EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                        CustomerPostingGroup.Validate("Payment Disc. Credit Acc.", AccountNumber);
+                    end;
+
+                    // Payment Tolerance Debit Acc.
+                    // Payment Tolerance Credit Acc.
+                    AccountNumber := HelperFunctions.GetGPAccountNumberByIndex(GPRM00201.RMWRACC);
+                    if AccountNumber <> '' then begin
+                        HelperFunctions.EnsureAccountHasGenProdPostingAccount(AccountNumber);
+                        CustomerPostingGroup.Validate("Payment Tolerance Debit Acc.", AccountNumber);
+                        CustomerPostingGroup.Validate("Payment Tolerance Credit Acc.", AccountNumber);
+                    end;
+
+                    CustomerPostingGroup.Insert();
+                end;
+        until GPRM00201.Next() = 0;
+
+        // Assign the Customer Posting Groups to the Customers
+        repeat
```

Comment:
Suggestion - could we move this code to the place where we are creating customers? We should ensure that Posting Groups are created before. 

It would save us one loop on the customers and 1 modify query on the customer.